docs/tool/js/views/filter-view.js

```diff
@@ -978,7 +979,19 @@ var filterView = {
         }
         filterView.indicateActivatedFilters();
     },
-
+    clearLocationBasedFilters: function(){
+        function isLocationBased(filterControl){
+            return (
+                filterControl.component.hasOwnProperty('zones') || 
+                filterControl.component.component_type === 'location'
+            );
+        }
+        for(var i = 0; i < filterView.filterControls.length; i++){
+            if(isLocationBased(filterView.filterControls[i]) && filterView.filterControls[i].isTouched()){
```

Comment:
This is currently triggering every locationbased control, whether or not it was touched. With this we're having the performance problem of recalculating the filtered data 20+ times.